src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_validators.py

```diff
@@ -4,24 +4,27 @@
 # --------------------------------------------------------------------------------------------
 
 from datetime import datetime
+from decimal import Decimal
 from azure.cli.core.util import CLIError
 
 
-def get_datetime_type():
+def datetime_type(string):
     """ Validates UTC datetime. Examples of accepted forms:
     2017-12-31T01:11:59Z,2017-12-31T01:11Z or 2017-12-31T01Z or 2017-12-31 """
-    def datetime_type(string):
-        """ Validates UTC datetime. Examples of accepted forms:
-        2017-12-31T01:11:59Z,2017-12-31T01:11Z or 2017-12-31T01Z or 2017-12-31 """
-        accepted_date_formats = ['%Y-%m-%dT%H:%M:%SZ', '%Y-%m-%dT%H:%MZ',
-                                 '%Y-%m-%dT%HZ', '%Y-%m-%d']
-        for form in accepted_date_formats:
-            try:
-                return datetime.strptime(string, form)
-            except ValueError:
-                continue
-        raise ValueError("Input '{}' not valid. Valid example: 2017-02-11T23:59:59Z".format(string))
-    return datetime_type
+    accepted_date_formats = ['%Y-%m-%dT%H:%M:%SZ', '%Y-%m-%dT%H:%MZ', '%Y-%m-%dT%HZ', '%Y-%m-%d']
+    for form in accepted_date_formats:
```

Comment:
Try the version and core. If you feel it doesn't support you scenarios, let me know so I can improve it.